viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if title != None and genre != None and rating != None:
+        movie_dict = {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data = add_to_watched(user_data, movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    watched_movies_list = []
+    sum = 0
+    for item in user_data["watched"]:
+        watched_movies_list.append(item["rating"])
+    for number in watched_movies_list:
+        sum += number

```

Comment:
Nice; we also could've used the built-in `sum` function to sum up everything in this list by `sum(watched_movies_list)`